### PR TITLE
Test not found exception (EXPOSUREAPP-6870)

### PR DESCRIPTION
**Pull request by mtwalli**
From the logs seems like multiple `CoronaTestNotFoundException` were thrown in both `SubmissionDeletionWarningViewModel` and `ShareTestResultNotificationService`
1- `SubmissionDeletionWarningViewModel` seems to catch the exceptions -> Error messages from VM is logged 
2- `catch` clause in `ShareTestResultNotificationService` does not seem to catch all crashes:
For example :
```kotlin
2021-04-28T21:04:54.452Z  V/ShareTestResultNotificationService$setup: Share positive test result notification counter has been reset for all test types
2021-04-28T21:04:54.456Z  V/CoronaTestRepository:HD: Data modifying failed (hasErrorHandler=false)
de.rki.coronawarnapp.coronatest.errors.CoronaTestNotFoundException: No found for qrcode-RAPID_ANTIGEN-cc265321af59e5e743aaa018c77a8ee5045cbf33e5d7162a61879825ae403481
	at de.rki.coronawarnapp.coronatest.CoronaTestRepository$removeTest$2.invokeSuspend(CoronaTestRepository.kt:18)
	at de.rki.coronawarnapp.coronatest.CoronaTestRepository$removeTest$2.invoke(CoronaTestRepository.kt:2)
	at de.rki.coronawarnapp.util.flow.HotDataFlow$internalProducer$1$invokeSuspend$$inlined$collect$1.emit(Collect.kt:10)
	at kotlinx.coroutines.flow.SharedFlowImpl.collect(SharedFlow.kt:12)
	at kotlinx.coroutines.flow.SharedFlowImpl$collect$1.invokeSuspend(Unknown Source:12)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:3)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:15)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:1)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:13)

2021-04-28T21:04:54.461Z  V/CoronaTestRepository:HD: Data modifying failed (hasErrorHandler=false)
de.rki.coronawarnapp.coronatest.errors.CoronaTestNotFoundException: No found for qrcode-RAPID_ANTIGEN-cc265321af59e5e743aaa018c77a8ee5045cbf33e5d7162a61879825ae403481
	at de.rki.coronawarnapp.coronatest.CoronaTestRepository$removeTest$2.invokeSuspend(CoronaTestRepository.kt:18)
	at de.rki.coronawarnapp.coronatest.CoronaTestRepository$removeTest$2.invoke(CoronaTestRepository.kt:2)
	at de.rki.coronawarnapp.util.flow.HotDataFlow$internalProducer$1$invokeSuspend$$inlined$collect$1.emit(Collect.kt:10)
	at kotlinx.coroutines.flow.SharedFlowImpl.collect(SharedFlow.kt:12)
	at kotlinx.coroutines.flow.SharedFlowImpl$collect$1.invokeSuspend(Unknown Source:12)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:3)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:15)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:1)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:13)
```

- Improve logging
- Remove finally refreshing logic  from `SubmissionTanViewModel` and `SubmissionDeletionWarningViewModel`

### Testing 
- Register different tests , delete - replace (Be creative :D )

### CODE REVIEW

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/coronatest/notification/ShareTestResultNotificationService.kt [hunk 1]
```diff
@@ -27,48 +26,20 @@ class ShareTestResultNotificationService @Inject constructor(
...
-
-                // cancel the reminder when test is submitted
-                tests
-                    .filter { it.isSubmitted }
-                    .forEach { notification.cancelSharePositiveTestResultNotification(it.type) }
-            }
-            .catch { Timber.e(it, "Failed to schedule positive test result reminder.") }
-            .launchIn(appScope)
-
+        // Schedule positive test reminders
```
Comment (LukasLechnerDev) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/coronatest/notification/ShareTestResultNotificationService.kt:29]: Is this comment necessary? Function name is already very descriptive :)

**Review by LukasLechnerDev — APPROVED** [review]
lgtm

**Review by BMItr — APPROVED** [review]
lgtm

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/coronatest/notification/ShareTestResultNotificationService.kt (hunk 1 above)
Comment (mtwalli) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/coronatest/notification/ShareTestResultNotificationService.kt:29]: Addressed in https://github.com/corona-warn-app/cwa-app-android/pull/3074

**Pull request merged**